src/lib/pmd/PmdCatalogWrapper.ts

```diff
@@ -10,43 +9,45 @@ import * as PmdLanguageManager from './PmdLanguageManager';
 import {PMD_LIB, PMD_VERSION, PmdSupport} from './PmdSupport';
 import path = require('path');
 import {uxEvents} from '../ScannerEvents';
+import { Controller } from '../../ioc.config';
+import { PMD_CATALOG_FILE } from '../../Constants';
 
 Messages.importMessagesDirectory(__dirname);
 const messages = Messages.loadMessages('@salesforce/sfdx-scanner', 'EventKeyTemplates');
 
 // Here, current dir __dirname = <base_dir>/sfdx-scanner/src/lib/pmd
 const PMD_CATALOGER_LIB = path.join(__dirname, '..', '..', '..', 'dist', 'pmd-cataloger', 'lib');
 const MAIN_CLASS = 'sfdc.sfdx.scanner.pmd.Main';
-const PMD_CATALOG_FILE = 'PmdCatalog.json';
-
 
 export class PmdCatalogWrapper extends PmdSupport {
 	private outputProcessor: OutputProcessor;
 	private logger: Logger; // TODO: add relevant trace logs
 	private initialized: boolean;
+	private sfdxScannerPath: string;
 
 	protected async init(): Promise<void> {
 		if (this.initialized) {
 			return;
 		}
 		this.logger = await Logger.child('PmdCatalogWrapper');
 		this.outputProcessor = await OutputProcessor.create({});
+		this.sfdxScannerPath = Controller.getSfdxScannerPath();
 
 		this.initialized = true;
 	}
 
 	public async getCatalog(): Promise<Catalog> {
 		this.logger.trace(`Populating PmdCatalog JSON.`);
 		await this.runCommand();
-		return PmdCatalogWrapper.readCatalogFromFile();
+		return this.readCatalogFromFile();
 	}
 
-	private static getCatalogPath(): string {
-		return path.join(SFDX_SCANNER_PATH, PMD_CATALOG_FILE);
+	private getCatalogPath(): string {
```

Comment:
I've made some static methods non-static to access `this`. Since we are registering these as Singletons in ioc and using injection to get values, I believe this shouldn't make a difference - please let me know if you think otherwise.